### DISCUSSION TITLE
[Urgent]Values set using reset are overridden with the values in useEffect

### DISCUSSION THREAD

**Discussion by nanivijay** [react-hook-form/react-hook-form#4734] [Q&A]
Repro steps:
https://codesandbox.io/s/react-hook-form-js-forked-oc3pn?file=/src/App.js

1. If you select "Category A" from the dropdown , I am showing firstName with value **"first name - useEffect"**
2. If I click on Load data after the above step the values I pass in the reset are applied successfully.
3. Click on refresh button
![image](https://user-images.githubusercontent.com/5945591/114361852-8314a780-9b2b-11eb-90b1-4f95cba436ec.png)

4. Now click on the load data button without selecting any thing in the dropdown

Expected result:
Dropdown should be selected to "Category A" and the first name should be **"first name - reset"**

Actual result:
Dropdown option selected to "Category A" but the first name is **"first name - useEffect"**

Could you please help how can I solve this issue?

This is blocking the add/edit form I developed in my project.

**Comment by bluebill1049** [comment]
> Refresh the codebox example ? 

hook form is not persisting your data selection, you should consider including that in a persist state management lib.

**Reply by nanivijay** [reply]
what do you mean by persist state management lib?

**Reply by bluebill1049** [reply]
when you refresh the page, the state is lost.

**Reply by nanivijay** [reply]
sorry for the confusion. Refresh means  reload the browser in the code sandbox example and click on the load data button to see the issue i am facing
